x/btccheckpoint/keeper/grpc_query.go

```diff
@@ -62,34 +62,33 @@ func (k Keeper) BtcCheckpointsInfo(ctx context.Context, req *types.QueryBtcCheck
 		return nil, status.Error(codes.InvalidArgument, "invalid request")
 	}
 
+	ckptInfoList := []*types.BTCCheckpointInfoResponse{}
 	epochDataStore := k.epochDataStore(ctx)
-
-	ckptInfoList := []*types.BTCCheckpointInfo{}
 	// iterate over epochDataStore, where key is the epoch number and value is the epoch data
 	pageRes, err := query.Paginate(epochDataStore, req.Pagination, func(key, value []byte) error {
-		epochNum := sdk.BigEndianToUint64(key)
 		var epochData types.EpochData
-		k.cdc.MustUnmarshal(value, &epochData)
+		if err := k.cdc.Unmarshal(value, &epochData); err != nil {
```

Comment:
```suggestion
		k.cdc.MustUnmarshal(value, &epochData)
```
Here `value` is retrieved from KV store thus has passed all the verifications. If it fails to be unmarshaled then something really bad happens (e.g., bugs in Babylon) and the DB state is corrupted. So instead of returning an error here maybe it makes more sense to panick directly. Wdyt?

Reply:
Have thought twice on this and I think you are right -- It does not make much sense to panick upon a single query, even if this signals a programming error. If unmarshaling epoch data has this programming error the node will have trouble somewhere else either way (e.g., in BeginBlock/EndBlock). So let's proceed with returning an error here :+1: